datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -321,10 +318,13 @@ pub(crate) struct SortPreservingMergeStream {
     next_batch_id: usize,
 
     /// min heap for record comparison
-    min_heap: BinaryHeap<SortKeyCursor>,
+    max_heap: BinaryHeap<Reverse<SortKeyCursor>>,
```

Comment:
This was a somewhat amusing surprise, `BinaryHeap` is a max heap, not a min heap, the comparator for `SortKeyCursor` was just backwards.

Reply:
But w/ `Reverse`, it's a "min heap" again, so I think the variable name should read `min_heap`.

Reply:
It's a max heap of reversed elements no?

Reply:
Well, we get into philosophical discussions here, but IMHO the variable should describe the entire construct (`BinaryHeap<Reverse<SortKeyCursor>>`), not just the outer shell (`BinaryHeap<...>`).

Should you decide the keep the name, then at least adjust the docstring which still read `min heap`.